src/_pytest/pathlib.py

```diff
@@ -226,11 +226,11 @@ def make_numbered_dir(root: Path, prefix: str) -> Path:
         else:
             _force_symlink(root, prefix + "current", new_path)
             return new_path
-    else:
-        raise OSError(
-            "could not create numbered dir with prefix "
-            "{prefix} in {root} after 10 tries".format(prefix=prefix, root=root)
-        )
+
```

Comment:
this one seems potentially problematic

Reply:
I think it's fine, no? The for-loop will always run (`range(10)` is never empty), and in the for-loop, either we succeed and `return`, or we fail 10 times and land here.